environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldSEBeanRegistrant.java

```diff
@@ -203,4 +213,166 @@ public boolean isNullable() {
         }
 
     }
+
+    void observesPATofDeprecatedActivator(@Observes @WithAnnotations(ActivateThreadScope.class) ProcessAnnotatedType<?> event) {
+        final AnnotatedType<?> originalAT = event.getAnnotatedType();
+        WeldSELogger.LOG.deprecatedActivatorAnnotationUsed(ActivateThreadScope.class.getName(),
+                org.jboss.weld.environment.se.contexts.activators.ActivateThreadScope.class.getName());
+        event.setAnnotatedType(new AnnotatedTypeWrapper(originalAT));
+    }
+}
+
+class AnnotatedTypeWrapper implements AnnotatedType {
+
+    private final AnnotatedType originalAT;
+    private final Set<? extends AnnotatedMethod<?>> annotatedMethods;
+    private final Set<Annotation> annotations;
+
+    public AnnotatedTypeWrapper(AnnotatedType annotatedType) {
+        this.originalAT = annotatedType;
+        this.annotatedMethods = annotatedType.getMethods();
+        this.annotations = originalAT.getAnnotations();
+    }
+
+    @Override
+    public Class<?> getJavaClass() {
+        return originalAT.getJavaClass();
+    }
+
+    @Override
+    public Set<? extends AnnotatedConstructor<?>> getConstructors() {
+        return originalAT.getConstructors();
+    }
+
+    @Override
+    public Set<? extends AnnotatedMethod<?>> getMethods() {
+        Set<AnnotatedMethod<?>> newMethods = new HashSet<>();
+        newMethods.addAll(annotatedMethods);
+
+        AnnotatedMethod<?> methodWithDeprecatedAnnotation = null;
+        AnnotatedMethod<?> newAnnotatedMethod = null;
+        for (final AnnotatedMethod<?> originalMethod : annotatedMethods) {
+            if (originalMethod.isAnnotationPresent(ActivateThreadScope.class)) {
+                methodWithDeprecatedAnnotation = originalMethod;
+                newAnnotatedMethod = new AnnotatedMethodWrapper(originalMethod);
+            }
+        }
+        if (methodWithDeprecatedAnnotation != null) {
```

Comment:
This will not work if more than one method is annotated with deprecated binding, right?